test/digraph-cyclicity-hadwick-james.jl

```diff
@@ -1,31 +1,58 @@
 @testset "Hadwick-James Cycles" begin
-    hadwick_james_ex1 = DiGraph(5)
-    add_edge!(hadwick_james_ex1, 1, 2)
-    add_edge!(hadwick_james_ex1, 2, 3)
-    add_edge!(hadwick_james_ex1, 3, 4)
-    add_edge!(hadwick_james_ex1, 3, 5)
-    add_edge!(hadwick_james_ex1, 4, 5)
-    add_edge!(hadwick_james_ex1, 5, 2)
 
-    hadwick_circuits = simplecycles_hadwick_james(hadwick_james_ex1)
+    # Simple graph gives expected circuits
+    ex1 = DiGraph(5)
+    add_edge!(ex1, 1, 2)
+    add_edge!(ex1, 2, 3)
+    add_edge!(ex1, 3, 4)
+    add_edge!(ex1, 3, 5)
+    add_edge!(ex1, 4, 5)
+    add_edge!(ex1, 5, 2)
 
     expected_circuits = Vector{Int}[
         [2, 3, 4, 5],
         [2, 3, 5]
     ]
-    @test issubset(expected_circuits, hadwick_circuits)
-    @test issubset(hadwick_circuits, expected_circuits)
+    for g in testdigraphs(ex1)
+        ex1_circuits = simplecycles_hadwick_james(g)
 
-    add_edge!(hadwick_james_ex1, 1, 1)
-    add_edge!(hadwick_james_ex1, 3, 3)
+        @test issubset(expected_circuits, ex1_circuits)
+        @test issubset(ex1_circuits, expected_circuits)
 
-    hadwick_circuits_self = simplecycles_hadwick_james(hadwick_james_ex1)
+        # Adding self-edges gives new circuits
+        add_edge!(g, 1, 1)
+        add_edge!(g, 3, 3)
 
-    @test issubset(expected_circuits, hadwick_circuits_self)
-    @test [1] ∈ hadwick_circuits_self && [3] ∈ hadwick_circuits_self
+        ex1_circuits_self = simplecycles_hadwick_james(g)
 
+        @test issubset(expected_circuits, ex1_circuits_self)
+        @test [1] ∈ ex1_circuits_self && [3] ∈ ex1_circuits_self
+    end
+
+    # Path DiGraph
     ex2_size = 10
-    hadwick_james_ex2 = PathDiGraph(ex2_size)
-    @test isempty(simplecycles_hadwick_james(hadwick_james_ex2))
+    ex2 = testdigraphs(PathDiGraph(ex2_size))
+    for g in ex2
+        @test isempty(simplecycles_hadwick_james(g))
+    end
+
+    # Complete DiGraph
+    ex3_size = 5
+    ex3 = testdigraphs(CompleteDiGraph(ex3_size))
+    for g in ex3
+        ex3_circuits = simplecycles_hadwick_james(g)
+        @test length(ex3_circuits) == length(unique(ex3_circuits))
+    end
 
+    # Almost fully connected DiGraph
+    ex4 = DiGraph(9)
+    for (src, dest) in [(1,2),(1,5),(1,7),(1,8),(2,9),(3,4),(3,6),
```

Comment:
Credit due to https://github.com/nchaudhr for this test case. "Simple" test graphs don't hit the branch in `unblock!`